### DISCUSSION TITLE
Which integrator should I use if I want to deal with transparent object?

### DISCUSSION THREAD

**Discussion by ChauChorHim** [mitsuba-renderer/mitsuba3#270] [Q&A] [answered]
Hi, I have some images of a transparent object captured in a darkroom and the only light source is a LCD screen with some patterns. I want to leverage the loss between these captured images and mitsuba rendered images to optimize the shape of transparent object. The question is thata 'path' integrator which seems not support to produce gradients of images with respect to the transparent object vertex positions. 'prb_reparam' and 'ptracer' are also not suitable for a transparent object (not a slab like the tutorial of caustics optimization). Could you please tell me which integrator is fitted my requirements? Thanks :)

**Comment by ziyi-zhang** [comment, marked as answer]
Hi @ChauChorHim, 
That reminds me of [this research](https://dl.acm.org/doi/pdf/10.1145/3414685.3417815) and a few related ones.
Assuming you are applying Mitsuba in their setup: homogenous glass, triangle mesh object, LCD patterned screen (we may be able to lift the constraint that a ray only interacts twice), 
- `prb` won't be necessary as the light path is short.
- `reparam` is useless in this setting as we don't have occlusion driven discontinuity.
- `ptracer` must be used in the caustic design because we have a directional emitter. Since you have a LCD monitor, this is no longer mandatory.
- `path` (the version in `path.cpp`) is a detached estimator. As you said, only the `eval_pdf` is attached to the autodiff tape.

Depending on your experiment setup, in my opinion, `ptracer` or a custom attached `path` might (I am not certain though =) ) be enough.

**Reply by ChauChorHim** [reply]
Hi Ziyi,

Thanks for your quick reply and detailed explanation. I will try to use ptracer then.

P.S. It is this paper you mentioned I am following now.
